daemon/cluster/services.go

```diff
@@ -260,13 +263,51 @@ func (c *Cluster) ServiceLogs(ctx context.Context, input string, config *backend
 		stdStreams = append(stdStreams, swarmapi.LogStreamStderr)
 	}
 
+	// Get tail value squared away - the number of previous log lines we look at
+	var tail int64
+	if config.Tail == "all" {
+		// tail of 0 means send all logs on the swarmkit side
+		tail = 0
+	} else {
+		t, err := strconv.Atoi(config.Tail)
+		if err != nil || t < 0 {
+			// TODO(dperny) better error message
+			return errors.New("tail relative to start of logs not supported")
```

Comment:
(I'm sorry, I mean it is legitimately a bad error message and I do need help getting it better, and it would benefit from some light bikeshedding to get right)